### ISSUE TITLE
Mobile focus order is incorrect

### ISSUE THREAD

**Issue by jamesjacobs** [jadu/pulsar#1140]
The focus order of the elements in the `header.toolbar` on mobile isn't ideal. As there isn't any relationship between the menu button, dropdowns and search form I'm not sure it's a strict failure (under 2.4.3 focus order or 1.3.2: Meaningful Sequence) but does warrant investigation.

Current focus order:

1. skip to content
2. menu button
3. search input
4. user dropdown
5. notifications

Ideal focus order:

1. skip to content
2. menu button
3. user dropdown
4. notifications
5. search input
